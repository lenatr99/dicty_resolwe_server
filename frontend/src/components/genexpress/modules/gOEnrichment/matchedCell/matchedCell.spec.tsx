import React from 'react';
import { fireEvent, screen } from '@testing-library/react';
import { Mock, vi } from 'vitest';
import GOEnrichmentMatchedCell from './matchedCell';
import { customRender } from 'tests/test-utils';
import { generateGOEnrichmentRow } from 'tests/mock';
import { GOEnrichmentRow } from 'redux/models/internal';

describe('gOEnrichmentMatchedCell', () => {
    let goEnrichmentRow: GOEnrichmentRow;
    let asFragment: () => DocumentFragment;
    let mockedOnMatchedGenesClick: Mock;
    const value = 12;

    beforeEach(() => {
        mockedOnMatchedGenesClick = vi.fn();
        goEnrichmentRow = generateGOEnrichmentRow(1);

        ({ asFragment } = customRender(
            <GOEnrichmentMatchedCell
                value={value}
                data={goEnrichmentRow}
                onMatchedGenesClick={mockedOnMatchedGenesClick}
            />,
        ));
    });

    it('should render as default snapshot', () => {
        expect(asFragment()).toMatchSnapshot();
    });

    it('should call onMatchedGenesClick when user clicks on the link', () => {
        fireEvent.click(screen.getByText(`${value}/${goEnrichmentRow.total}`));

        expect(mockedOnMatchedGenesClick.mock.calls.length).toBe(1);
        expect(mockedOnMatchedGenesClick.mock.calls[0][0]).toBe(goEnrichmentRow);
    });
});
