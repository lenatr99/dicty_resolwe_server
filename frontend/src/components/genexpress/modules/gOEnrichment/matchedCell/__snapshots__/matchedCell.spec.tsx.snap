// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`gOEnrichmentMatchedCell > should render as default snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-jCbFiK hkFjrU"
  >
    <button
      class="MuiButtonBase-root-ocQuz kiSjtm MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textSecondary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorSecondary MuiButton-fullWidth MuiButton-root-jMlkZG hGfmia MuiButton-root MuiButton-text MuiButton-textSecondary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorSecondary MuiButton-fullWidth sc-cBYhjr hXKyFk"
      tabindex="0"
      type="button"
    >
      12/250
      <span
        class="MuiTouchRipple-root-huKFMv dxeyKA MuiTouchRipple-root"
      />
    </button>
  </div>
</DocumentFragment>
`;
